test/common/stats/statsd_test.cc

```diff
@@ -54,13 +55,15 @@ class TcpStatsdSinkTest : public testing::Test {
 TEST_F(TcpStatsdSinkTest, All) {
   InSequence s;
 
-  expectCreateConnection();
-  EXPECT_CALL(*connection_, write(BufferStringEqual("envoy.test_counter:1|c\n")));
+  sink_->beginFlush();
```

Comment:
What about testing an empty flush, i.e. back-to-back `beingFlush(); endFlush()`.